src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSegmentEntry.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using RegressionGames;
+using RegressionGames.StateRecorder;
+using RegressionGames.StateRecorder.BotSegments;
+using RegressionGames.StateRecorder.BotSegments.Models;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+/**
+ * <summary>
+ * Displays the high-level information for a Segment, and enables the user to play it
+ * </summary>
+ */
+public class RGSegmentEntry : MonoBehaviour
+{
+    public string segmentName;
+
+    public string description;
+    
+    // will be a file path when in-editor, or a resources path when in a packaged build
+    public string path;
+
+    public BotSequenceEntryType type;
+    
+    /**
+     * UI component fields
+     */
+    [SerializeField]
+    public Button playButton;
+    
+    [SerializeField]
+    public TMP_Text nameComponent;
+
+    [SerializeField]
+    public TMP_Text descriptionComponent;
+    
+    [SerializeField]
+    public GameObject segmentListIndicatorComponent;
+
+    /**
+     * <summary>
+     * Ensures all needed prefabs and fields are properly set. Shows an icon if this segment is of the SegmentList type
+     * </summary>
+     */
+    public void Start()
+    {
+        if (string.IsNullOrEmpty(path))
+        {
+            Debug.LogError($"RGSegmentEntry is missing its path");

```

Comment:
(non-blocking) At some point we should do away with these checks and error log messages.  These are SDK breaking conditions that should really throw exceptions, not just log a message and continue

Reply:
Fair point. I've removed any of these checks I've added in this PR from the relevant `Start` methods. I also removed these from the Sequence editor